internal/gitutil/gitutil.go

```diff
@@ -32,10 +32,34 @@ import (
 // for remote repos.  Defaults to UserHomeDir/.kpt/repos if unspecified.
 const RepoCacheDirEnv = "KPT_CACHE_DIR"
 
+var DefaultRef = "main"
+
+// SetDefaultRef sets(and returns) the DefaultRef to "master" if master branch exists in
+// remote repository, falls back to "main" if master branch doesn't exist
+func SetDefaultRef(repo string) string {
+	gitProgram, err := exec.LookPath("git")
+	if err != nil {
+		return DefaultRef
+	}
+	out := bytes.Buffer{}
+	cmd := exec.Command(gitProgram, "ls-remote", repo, "master")
+	cmd.Stdout = &out
+	err = cmd.Run()
+	if err != nil {
```

Comment:
Is the only reason we can get an error here that the "master" branch doesn't exist?

Reply:
Added a comment about the rationale.

Reply:
I still don't follow this. Based on the comment, it seems like there is a chance the error is os-related, in which case we should return the error.

Reply:
Is the comment here still correct?

Reply:
Maybe add more comments in this function to explain what we expect to find in stderr/stdout after running the git command.